src/subprocess-posix.cc

```diff
@@ -90,9 +88,12 @@ bool Subprocess::Start(SubprocessSet* set, const string& command) {
         // Now can use stderr for errors.
         error_pipe = 2;
         close(output_pipe[1]);
+      } else {
+        // In the console case, the pipe is not needed by the child
+        // process at all. Close it in the child, but keep it open
+        // in the parent in case execl fails.
+        SetCloseOnExec(output_pipe[1]);
```

Comment:
Ah, I think I see. I'll paraphrase below, please let me know if I misunderstand.

DoWork/OnPipeReady is using the pipe to determine whether or not the child is done. If there were a console pool task, it would immediately go do Done, and then Finish would block until it's really done. While blocking, other tasks' buffers would fill up, potentially blocking them from doing work.

For *only* console pool tasks, DoWork could instead use waitpid. That would have the same effect as ppoll on a non-console task that never sends anything to stdout (i.e. they'd both wait until the subprocess dies).